packages/Frontend/__mocks__/firebase.js

```diff
@@ -0,0 +1,10 @@
+const firebase = jest.genMockFromModule('firebase');
```

Comment:
This is a mock of firebase, it's setup so that no test that we write ever actually tries to log in to firebase for real. You can read more about jest mocks here: https://facebook.github.io/jest/docs/en/manual-mocks.html#mocking-node-modules

The reason you would want to mock some things for tests is to isolate what you're testing. The tests that we've written so far are called unit tests, they're meant to test a small unit of code, usually a function, or maybe a class. They're not mean to test how that function or class interacts with other functions or classes.

Basically, with firebase, we don't want to test with the real firebase, because maybe firebase happens to be a little slow that day, or it messes up one of the requests even though 99.99% of the time it handles them right. Our test would fail even though the code is just fine, it's the thing it's interacting with that failed.

There are tests that are meant to see how things interact with other things in code. There are integration tests and end to end tests that handle those sort of things. We haven't written any of those yet, because our app is relatively small at this point, but we will.

Usually, the tests of your application are largely unit tests, then integration tests, then end to end tests. Jest should be able to handle our integration tests when we get to them, but for end to end tests we'll probably use nightwatch from http://nightwatchjs.org/. It allows you to control a browser with javascript, that way we can write code that will visit a page, click a button, wait for the next page to load, write some text in, save it, visit another page, etc.